### ISSUE TITLE
Investigate replay-based rehydration

### ISSUE THREAD

**Issue by mridgway** [yahoo/dispatchr#22]
Theoretically, stores could be reset to their original state by recording the actions and payloads that were dispatched on the server, serializing them, and then replaying them on the client to rehydrate the application. An investigation on the performance and implementation benefits for each should be done.

This could reduce the amount of code in each store since they would no longer need to implement dehydrate/rehydrate functions.

**Comment by stanshillis** [comment]
Rehydrtaion of stores through replay will bring it much closer to event sourcing ideas. We are currently experimenting with replay and it works well. There are only a handful of events being replayed so performence has not been an issue. It would definitely be great to see support for that directly in the library.

**Comment by mridgway** [comment]
Yeah, I think it depends on the application, so it could be a configuration option. The implementation should be fairly simple. I'm totally willing to add this in if someone wants to do a PR.

**Comment by ptomasroos** [comment]
If the dehydrate / rehydrate methods won't go away what's the purpose ?

The goal was to reduce code right so in that case I would say there should be a single way to do the state mgmt on client side init. 

But hey I'm a purist and like simplicity.

**Comment by mridgway** [comment]
The need for each store to define dehydrate/rehydrate would go away. With replay, it would just re-dispatch the events on the client.

**Comment by ptomasroos** [comment]
Right, It felt like you wanted both. I could def see the benefit by having a self solving rehydration.

**Comment by 3den** [comment]
Could we add an standard dehydrate / rehydrate to the BaseStore? Those methods could just set/get `this.state` which is what is done in most cases.

**Comment by ptomasroos** [comment]
Elegant idea. Either that or a mixin would work.
